### ISSUE TITLE
New assertions: File/Path.shouldHaveNameWithoutExtension

### ISSUE THREAD

**Issue by sksamuel** [kotest/kotest#1696]
**Please describe the feature you'd like to see including any solutions in mind if you have any**

**Comment by LeoColman** [comment]
I would shorten this to `shouldHaveExtension` + `shouldNotHaveExtension`, what do you think?

**Comment by sksamuel** [comment]
Different semantics.


"myfile.json".shouldHaveNameWithoutExt("myfile")

**Comment by LeoColman** [comment]
Ah, I thought it was supposed to be
```
"myfile.json".shouldHaveNameWithoutExtension()
```

**Comment by sksamuel** [comment]
Released in 4.2.5
